.github/workflows/ci_cd.yml

```
@@ -23,110 +25,84 @@ jobs:
+        uses: pyansys/actions/code-style@v1
         with:
           python-version: ${{ env.MAIN_PYTHON_VERSION }}
 
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip flit tox
-
-      - name: Test with tox
-        run: tox -e style
-
   docs-style:
-    name: Documentation Style Check
+    name: PyAnsys documentation style check
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-  
-      - name: Running Vale
-        uses: errata-ai/vale-action@reviewdog
-        env:
-          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+      - name: PyAnsys documentation style checks
+        uses: pyansys/actions/doc-style@v1
         with:
-          files: doc
-          reporter: github-pr-check
-          level: error
-          filter_mode: nofilter
-          fail_on_error: true
-          vale_flags: "--config=doc/.vale.ini"
+          token: ${{ secrets.GITHUB_TOKEN }}
 
-  tests:
-    name: Tests
-    needs: [style]
+  smoke-tests:
+    name: Build and smoke tests
     runs-on: ${{ matrix.os }}
+    needs: [style]
     strategy:
-      matrix:
-        os: [windows-latest, ubuntu-latest, macos-latest]
-        cfg:
-          - {python-version: "3.7", toxenv: "py37", toxargs: "cov"}
-          - {python-version: "3.8", toxenv: "py38", toxargs: "cov"}
-          - {python-version: "3.9", toxenv: "py39", toxargs: "cov"}
-          - {python-version: "3.10", toxenv: "py310", toxargs: "doctests-cov"}
       fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: ['3.7', '3.8', '3.9', '3.10']
 
     steps:
-      - name: Clone repository
-        uses: actions/checkout@v3
-
-      - name: Set up Python ${{ matrix.cfg.python-version }}
-        uses: actions/setup-python@v4
+      - name: Build wheelhouse and perform smoke test
+        uses: pyansys/actions/build-wheelhouse@v1
         with:
-          python-version: ${{ matrix.cfg.python-version }}
-
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip tox
-
-      - name: Test with tox
-        run: tox -e ${{ matrix.cfg.toxenv }}-${{ matrix.cfg.toxargs }}
+          library-name: ${{ env.PACKAGE_NAME }}
+          library-namespace: ${{ env.PACKAGE_NAMESPACE }}
+          operating-system: ${{ matrix.os }}
+          python-version: ${{ matrix.python-version }}
 
-      - name: Install package
-        run: |
-          pip install -r requirements/requirements_build.txt
-          pip install .
-
-      - name: Store version
-        run: |
-          echo "::set-output name=PYANSYS_TOOLS_VERSIONING_VERSION::$(python -c "from ansys.tools.versioning import __version__; print(__version__)")"
-        id: version
-
-      - name: Generate wheelhouse
-        run: |
-          pip wheel . -w wheelhouse
+  smoke-tests-macos:
+    name: Build and smoke tests (macOS)
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
```

I bet that, when using multiple assertions inside GitHub conditionals, those require from using the "dereference" operator:

```suggestion
    if: ${{ github.event_name == 'push' && contains(github.ref, 'refs/tags') }}
```

---

Its very confusing...